### DISCUSSION TITLE
er图中链接桩的配置

### DISCUSSION THREAD

**Discussion by wangyueAI023** [antvis/X6#2233] [Q&A] [answered]
1.
![image](https://user-images.githubusercontent.com/42490685/174219928-9bc71928-7c67-487e-b789-44ca475894e3.png)
怎么设置锚点和边之间的距离？想要的时果是连接链接桩后 边自动距离链接桩一定距离进行连接 空出滚动条的位置
2.
![image](https://user-images.githubusercontent.com/42490685/174234608-9faae46a-3d6f-4f06-aeae-64fd7e3eb5a1.png)
怎么设置移动节点时，连接的锚点自动切换到距离较近的锚点，想要的效果是，只进行同一行的两个锚点的互相切换

**Comment by NewByVector** [comment, marked as answer]
1. 可以使用 [nodeAnchor](https://x6.antv.vision/zh/docs/api/registry/node-anchor/#gatsby-focus-wrapper) 来实现
2. 第二个能否更具体点，或者翻译成技术需求是？

**Reply by wangyueAI023** [reply]
![微信图片_20220620125331](https://user-images.githubusercontent.com/42490685/174557251-3ad78416-7eb9-4386-939a-07ba751c0390.png)
2.开始为   节点A的链接桩b->节点B的链接桩a
移动A   改为节点A的链接桩a->节点B的链接桩b
仅允许同一行的链接桩(如ab)进行切换
如何设置这种切换?

**Reply by NewByVector** [reply]
可以使用 `node.getPortsPosition()` 拿到每个连接桩的位置，节点移动的时候，计算同一行距离最近的两个连接桩，然后修改之前连线，使它连接到新的连接桩。
